action.yaml

```diff
@@ -0,0 +1,367 @@
+      cat("::endgroup::\n")
+
+      cat("::group::Assertions\n")
+      if (isFALSE(file.exists(file.path(path, "DESCRIPTION")))) {
+        cli::cli_alert_danger("{.file DESCRIPTION} file not found in {.path {path}}. Skipping.")
+        return()
+      }
+      if (length(lookup_refs) == 0 || identical(lookup_refs, "")) {
+        cli::cli_alert_danger("No lookup refs provided. Skipping.")
+        return()
+      }
+      parsed_lookup_refs <- pkgdepends::parse_pkg_refs(lookup_refs)
+      if (any(!vapply(parsed_lookup_refs, is, logical(1), "remote_ref_github"))) {
+        incorrect_idx <- which(!vapply(parsed_lookup_refs, is, logical(1), "remote_ref_github"))
+        cli::cli_alert_danger("All lookup refs must be GH refs. Incorrect refs: {.val {lookup_refs[incorrect_idx]}}.")
+        return()
+      }
+      cat("::endgroup::\n")
+
+      cat("::group::Modify DESCRIPTION file\n")
+      d <- desc::desc(path)
+      desc_field_create(d, "Config/Needs/DepsDev")
+
+      dependency_types <- c(
+        pkgdepends::as_pkg_dependencies(TRUE)$direct,
+        "Config/Needs/DepsBranch",
+        "Config/Needs/DepsDev"
+      )
+
+      x <- pkgdepends::new_pkg_deps(path, config = list(dependencies = dependency_types))
+
+      while (TRUE) {
+        x$solve()
+
+        if (x$get_solution()$status == "OK") {
+          cli::cli_alert_success("Dependencies resolved successfully.")
+          break()
+        }
+
+        failed_pkgs <- x$get_solution()$failures[1, "failure_down"]
+        cli::cli_alert_warning("Failed to resolve dependencies. Failed packages: {.val {failed_pkgs}}.")
+
+        for (pkg in failed_pkgs) {
+          pkg_ref <- find_ref(pkg, lookup_refs)
+          if (length(pkg_ref) == 0) {
+            cli::cli_alert_danger("Package {.pkg {pkg}} not found in the {.code {lookup_refs}} list.")
+            break()
+          }
+          cli::cli_alert_info("Adding package {.pkg {pkg}} (ref: {.val {pkg_ref$ref}}) to the {.field Config/Needs/DepsDev} field.")
+          desc_field_append(d, "Config/Needs/DepsDev", pkg_ref$ref)
+          d$write(path)
+        }
+
+        cli::cli_alert_info("Trying to resolve dependencies again.")
+        x <- pkgdepends::new_pkg_deps(path, config = list(dependencies = dependency_types))
+      }
+
+      desc_field_clear_if_empty(d, "Config/Needs/DepsDev")
+      d$write(path)
+      cat("::endgroup::\n")
+
+      cat("::group::Print added field\n")
+      cat(d$get("Config/Needs/DepsDev"))
+      cat("\n")
+      cat("::endgroup::\n")
+
+      cat("::group::Print DESCRIPTION file\n")
+      d
+      cat("::endgroup::\n")
+
+  - uses: r-lib/actions/setup-r-dependencies@v2
+    if: ${{ inputs.skip-install == 'false' }}
+    with:
+      working-directory: ${{ inputs.repository-path }}
+      extra-packages: ${{ inputs.extra-packages }}
+      needs: |
+        ${{ steps.dep-types.outputs.needs }}
+        DepsBranch
+        DepsDev
+      dependencies: ${{ inputs.dependencies }}
```

Comment:
This parameter allows to specify the type(s) of dependencies to be used - all, hard, soft, none etc.
We need to either respect it in the deps-feature and deps-dev or remove it complitely.